docs/architecture/adr-021-subspace-specific-custom-fee-tokens.md

```diff
@@ -0,0 +1,109 @@
+2. Validators will agree or reject the proposal through on-chain voting. 
+3. If the proposal is accepted, the new token denom will be added to the list of additional fee token denoms that can be used to pay for fees within that subspace.
+4. If the proposal is rejected, the new token denom will not be added to the list of additional fee token denoms.
+
+### Types
+
+To represent the list of supported fee token denoms and their default minimum prices within a subspace, we will add a new field `allowed_fee_tokens` to the existing Subspace structure.
+
+```proto
+message Subspace {
+
+  ...skip
+
+  // the creation time of the subspace
+  google.protobuf.Timestamp creation_time = 7;
+  
+  // List of fee token denoms with default minimum gas prices allowed inside the subspace
+  repeated Coin allowed_fee_tokens = 8;
+}
+```
+
+### Msgs
+
+We will implement the operation that allows subspace admins to updates subspace allowed fee tokens list by governance;
+
+```proto
+service Msg {
+  // UpdateSubspaceFeeTokens allows subspace admins to update the allowed tokens to be fee tokens inside the subspace by governance
+  rpc UpdateSubspaceFeeTokens(MsgUpdateSubspaceFeeTokens) returns (MsgUpdateSubspaceFeeTokensResponse);
+}
+
+// MsgUpdateSubspaceFeeTokens represents the message to be used to update a subspace fee tokens by governance
+message MsgUpdateSubspaceFeeTokens {
+
+  // Id of the subspace where the list of allowed fee tokens will be updated
+  uint64 subspace_id = 1;
+    
+  // List of the allowed tokens to be fee token inside the subspace along with their default minimum prices
+  repeated Coin allowed_fee_tokens = 2;
+    
+  // authority is the address that controls the module (defaults to x/gov unless overwritten).
+  string authority = 3;
+}
+
+// MsgUpdateSubspaceFeeTokensResponse represents the Msg/UpdateSubspaceFeeTokens response type
+message MsgUpdateSubspaceFeeTokensResponse {}
+```
+
+### Custom TxFeeChecker
+
+To make it easier for validators to manage the minimum prices of fee tokens allowed within a subspace, we will develop a new `TxFeeChecker` based on the existing one, which will function as follows:
+- Combine the list of minimum gas prices in the validator's local configuration with the list of allowed fee tokens and their minimum prices within the subspace.
+- Follow the same process as the existing `TxFeeChecker`.
+
+### Deprecated `x/fees`
+
+Since `x/fees` module may conflict with the custom fee tokens feature, as demonstrated in the following scenario:
+- Governance decides to change the minimum fees of MsgCreatePost to `10dsm` using `x/fees`.
+- Governance accepts the custom token `minttoken` as a fee within subspace 1 using `MsgUpdateSubspaceFeeTokens`.
+
+Due to this conflict, `minttoken` will not be accepted as a fee within the subspace 1. Therefore, we decided to drop `x/fees` from Desmos.
```

Comment:
```suggestion
By implementing the mechanism described above, there might be a problem with the `x/fees` module. Here's an example scenario to illustrate this:

1. the governance decides to use `x/fees` to change the minimum fees of `MsgCreatePost` to `10dsm`.
2. the governance also wants to use a custom token called `TokenA` as a fee token within subspace `1` using `MsgUpdateSubspaceFeeTokens`.

Because `x/fees` takes priority over anything, `TokenA` will not be accepted as a fee token within subspace `1`. 

For this reason, we have decided to remove the `x/fees` module from Desmos.
```